runner/reporter.go

```diff
@@ -124,38 +121,37 @@ func newReporter(results chan *callResult, c *RunConfig) *Reporter {
 	cap := min(c.n, maxResult)
 
 	return &Reporter{
-		config:         c,
-		results:        results,
-		done:           make(chan bool, 1),
+		config:  c,
+		results: results,
+		done:    make(chan bool, 1),
+		details: make([]ResultDetail, 0, cap),
+
 		statusCodeDist: make(map[string]int),
 		errorDist:      make(map[string]int),
-		lats:           make([]float64, 0, cap),
 	}
 }
 
 // Run runs the reporter
 func (r *Reporter) Run() {
 	for res := range r.results {
+		errStr := ""
+		if res.err != nil {
+			errStr = res.err.Error()
+		}
+
 		r.totalCount++
-		r.avgTotal += res.duration.Seconds()
+		r.totalLatenciesSec += res.duration.Seconds()
+		r.statusCodeDist[res.status]++
 		if res.err != nil {
-			errStr := res.err.Error()
 			r.errorDist[errStr]++
```

Comment:
Can we please just put the line
```
errStr = res.err.Error()
```
into this if block so that we only have one error check block